clowder-framework/healthmonitor
# Refactoring healthmonitor

Changes:
- Refactored monolithic `healthmonitor.py` into smaller pieces

See [README.md](https://github.com/clowder-framework/healthmonitor/blob/refactoring-healthmonitor/README.md) for more details

How to Test:
1. Checkout and run this branch locally, rebuild the Docker image: `docker build -t clowder/healthmonitor .` 
2. Edit the `config.yml` to reflect the `checks` you want to run and the `notifiers` to run for reporting
    * See [README.md](https://github.com/clowder-framework/healthmonitor/blob/refactoring-healthmonitor/README.md) for more details on configuration
3. Run a container from the newly-built image: `docker run --rm -v $(pwd)/config.yml:/src/config.yml clowder/healthmonitor python healthmonitor.py`
    * You should see the healthmonitor(s) that you've configured are now running as separate threads
 4. Wait for awhile
    * You should see results (either failures or successes) coming through in the logs
